Formula/k/kuzu.rb

```diff
@@ -0,0 +1,57 @@
+class Kuzu < Formula
+  desc "Embeddable graph database management system built for query speed & scalability"
+  homepage "https://kuzudb.com/"
+  url "https://github.com/kuzudb/kuzu/archive/refs/tags/v0.4.1.tar.gz"
+  sha256 "e81be1e94dafba5e4a6f08b7f3a530c75926740d47e3a6b198687cd132630b6d"
+  license "MIT"
+  head "https://github.com/kuzudb/kuzu.git", branch: "master"
+
+  depends_on "cmake" => :build
+  depends_on "python@3.12" => :build
+
+  def install
+    args = %w[
+      -DAUTO_UPDATE_GRAMMAR=0
+      -DENABLE_LTO=1
+    ]
+
+    system "cmake", "-S", ".", "-B", "build", *args, *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+
+    bin.install bin/"kuzu_shell" => "kuzu"
```

Comment:
I think it should be technically symlink?

also, maybe change it in the cmakefile to create artifact as kuzu?

Reply:
Thanks for your review. I have changed it to symlink. We will consider renaming it in cmakefile for our next release.

Reply:
Sounds good, let us know when we can re-review.

Reply:
I think it is ready for re-review now. Our next stable release will be months away. Could we keep the `install_symlink` for now? I will remove this when I bump this formula after the next release.

Reply:
any PR or issue for tracking the binary artifact change?

Reply:
OK I created a new issue at our side: https://github.com/kuzudb/kuzu/issues/3458